File: r/high-scores/2/high-scores.R
scores_list <- function(scores) {
  scores
}

latest <- function(scores) {
  tail(scores, 1)
}

personal_best <- function(scores) {
  max(scores)
}

personal_top_three <- function(scores) {
  head(sort(scores, decreasing = TRUE), n = 3L)
}

File: python/high-scores/2/high_scores.py
class HighScores:
    def __init__(self, scores):
        self.score_list = scores

    def scores(self):
        return self.score_list

    def latest(self):
        return self.score_list[-1]

    def personal_best(self):
        return max(self.score_list)

    def personal_top_three(self):
        sorted_scores = sorted(self.score_list, reverse=True)
        return sorted_scores[:3] if len(sorted_scores) >= 3 else sorted_scores
